tensorboard/data/server/run.rs

```diff
@@ -145,13 +156,19 @@ impl StageTimeSeries {
     }
 }
 
+/// Minimum time to wait between committing while a run is still loading.
+const COMMIT_INTERVAL: Duration = Duration::from_secs(5);
+
 impl RunLoader {
-    pub fn new() -> Self {
+    pub fn new(run: Run) -> Self {
         Self {
-            start_time: None,
+            run: run,
             files: BTreeMap::new(),
-            time_series: HashMap::new(),
             checksum: true,
+            data: RunLoaderData {
+                start_time: None,
+                time_series: HashMap::new(),
+            },
```

Comment:
Optional: if you let `RunLoaderData` derive `Default`, then this is its
default value, so can be just `data: RunLoaderData::default()`.